src/Parser/Rogue/Subtlety/Modules/ResourceTracker/ResourceTracker.js

```diff
@@ -92,6 +92,12 @@ class ResourceTracker extends Analyzer {
 
     //Re-sync current amount, to update not-tracked gains.
     this.current = eventResource.amount - cost;
+
+    this.onSpent(cost);
+  }
+
+  onSpent(spent) {
```

Comment:
This is an interesting design, though I think it would be more sensibly implemented as a callback registry. Basically keep an array of `_onSpentCallbacks`, do something like 
```
onSpent(callback) {
  this._onSpentCallbacks.push(callback);
}
```
... and then when the module detects resources spent, it does `this._onSpentCallbacks.forEach(callback => callback(amount, event));` ... this pattern allowing other analyzers to register things that happen on resource spend, rather than having to pack it all into the ComboPointTracker / EnergyTracker.

Reply:
So Observable pattern? Sounds good, will change it when I work on it next. I imagine I can register observers in on_initialized ?